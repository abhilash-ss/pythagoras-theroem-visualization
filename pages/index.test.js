import React from "react";
import { shallow } from "enzyme";
import { withRouter } from "next/router";
import Home from "./index";

jest.mock("next/router", () => ({
  withRouter: (component) => component
}));

describe("Home", () => {
  let props;
  beforeEach(() => {
    props = {
      router: {
        push: jest.fn()
      }
    };
  });
  it("should render without crash", () => {
    const component = shallow(<Home {...props} />);
    expect(component).toMatchSnapshot();
  });

  it("should handle error case", () => {
    const component = shallow(<Home {...props} />);
    const errorMsg = "Enter the sides of triangles ang click to see";
    component
      .find("input")
      .at(0)
      .props()
      .onChange({ target: { value: "onChange" } });

    component
      .find("input")
      .at(1)
      .props()
      .onChange({ target: { value: "onChange" } });

    component
      .find("input")
      .at(2)
      .props()
      .onChange({ target: { value: "onChange" } });

    component.find("button").props().onClick();

    const text = component.find("p").text();
    expect(text).toEqual(errorMsg);
  });

  it("should handle success case", () => {
    const component = shallow(<Home {...props} />);
    component
      .find("input")
      .at(0)
      .props()
      .onChange({ target: { value: "3" } });

    component
      .find("input")
      .at(1)
      .props()
      .onChange({ target: { value: "4" } });

    component
      .find("input")
      .at(2)
      .props()
      .onChange({ target: { value: "5" } });

    component.find("button").props().onClick();
    expect(props.router.push).toHaveBeenCalled();
  });

  it("should handle success case", () => {
    const component = shallow(<Home {...props} />);
    component.find("button").props().onClick();
    expect(props.router.push).not.toHaveBeenCalled();
  });
});
