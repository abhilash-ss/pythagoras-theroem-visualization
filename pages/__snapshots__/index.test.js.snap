// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Home should render without crash 1`] = `ShallowWrapper {}`;
